scripts/migrate-portals.js

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env node
+const fs = require('fs');
+const readline = require('readline');
+const { MatrixClient, LogService } = require('matrix-bot-sdk');
+
+LogService.setLevel('ERROR');
+
+const client = new MatrixClient(process.env.MATRIX_URL, process.env.MATRIX_AT);
+
+const dryRun = process.env.DRY_RUN ? process.env.DRY_RUN === 'true' : true;
+
+const rl = readline.createInterface({
+    input: fs.createReadStream(process.env.ROOM_STORE || './room-store.db'),
+    output: process.stdout,
+    terminal: false
+});
+
+let portalRooms = [];
+
+rl.on('line', (line) => {
+    const roomEntry = JSON.parse(line);
+    if (!roomEntry.matrix_id) {
+        return;
+    }
+    if (!roomEntry || !roomEntry.data || !roomEntry.data.portal) {
```

Comment:
Technically in our plan we wanted to filter out rooms which were portals but ones that had been manually tamperted with. In hindsight, I do not think we should filter those out.

Reply:
Do we want to check that it is a `matrix.org` room?

Probably another good indicator whether it's a portal or plumbed room.

Reply:
can't hurt :)